source/blog/2017-07-18-running-kubernetes-on-fedora-atomic-26.html.md

```diff
@@ -0,0 +1,161 @@
+
+```bash
+# rpm-ostree install kubernetes-kubeadm ethtool ebtables
+```
+
+After installing, you either have to reboot (using `systemctl reboot` or by tacking an `-r` onto the end of the install command above) or you can apply the changes using the experimental command `rpm-ostree ex livefs`.
+
+In order for kubeadm to work with SELinux in enforcing mode, you'll need to [for now](https://github.com/kubernetes/kubeadm/issues/279) create the following directory and set its SELinux context as follows:
+
+```bash
+# mkdir /etc/kubernetes/pki
+
+# chcon -Rt container_share_t /etc/kubernetes/pki
+```
+
+Also, kubeadm requires a different restart-on-fail behavior from the kubelet, so we'll need to add three lines to this drop-in file (this step won't be necessary once [this commit](http://pkgs.fedoraproject.org/cgit/rpms/kubernetes.git/commit/?id=e1f50eb5233848580ed354b1ec8b0c886ce8caaf) makes its way into the stable kubeadm rpm):
+
+```bash
+# vi /etc/systemd/system/kubelet.service.d/kubeadm.conf
+
+Restart=always
+StartLimitInterval=0
+RestartSec=10
+
+# systemctl daemon-reload
+```
+
+From here, you can follow the [upstream kubeadm documentation](https://kubernetes.io/docs/setup/independent/create-cluster-kubeadm/) to bring up a cluster. Note, though, that you'll have to append `--skip-preflight-checks` to the `kubeadm init` command because kubeadm currently does not know where to look for Fedora's kernel module configuration. See this [pull request](https://github.com/kubernetes/kubernetes/pull/49410) for more information.
+
+Also, most of the network plugins I've tested with Kubeadm have an issue running with SELinux confinement, which is one of the reasons why the upstream docs suggest putting SELinux into permissive mode. There are a couple of ways to avoid disabling this security feature on your host, however. I typically edit the yaml file that configures the network plugin to tell Kubernetes to run the plugin as type [`spc_t`](http://danwalsh.livejournal.com/74754.html), which leaves its containers unconfined by SELinux.
+
+For instance, here's a portion of the Flannel plugin yaml that I've edited:
+
+```bash
+spec:
+  template:
+    metadata:
+      labels:
+        tier: node
+        app: flannel
+    spec:
+      securityContext:
+        seLinuxOptions:
+          type: "spc_t"
+      hostNetwork: true
+```
+
+The three lines beginning with `securityContext:` go in right before the `hostNetwork: true` line. This same trick should work in any of the network plugin yaml files.
+
+I've opened [an issue](https://pagure.io/atomic/kubernetes-sig/issue/3) here to track efforts to get SELinux-compatible changes into these upstream plugins. Head over there to track progress or help out.
+
+## Ansible Deployment
+
```

Comment:
should we mention the transition from kubeadm here. A more advanced install option to kubeadm is the ansible blah blah.